### ISSUE TITLE
Docs should discuss left-to-right evaluation

### ISSUE THREAD

**Issue by LiberalArtist** [racket/racket#4189]
While  [replying to a Discord question](https://racket.discourse.group/t/continuation-question-from-matt-mights-blogs/825/14?u=liberalartist), I couldn't find a great citation for the fact that Racket—in contrast to Scheme—specifies left-to-right evaluation. It's implicitly illustrated in §§ [1.1.1 Sub-expression Evaluation and Continuations](https://docs.racket-lang.org/reference/eval-model.html#%28part._cont-model%29) and [1.1.7 Procedure Applications and Local Variables](https://docs.racket-lang.org/reference/eval-model.html#%28part._.Procedure_.Applications_and_.Local_.Variables%29), but it seems worth discussing explicitly in one of those places. I remember other people having wondered about it, especially savvy Schemers who may have been bitten by the RNRS reports leaving it unspecified.

**Comment by mflatt** [comment]
I agree that it could be mentioned in more places, but I'll note that left-to-right evaluation is specifically a property of the `#%app` form, and it's specified [there](https://docs.racket-lang.org/reference/application.html#%28form._%28%28lib._racket%2Fprivate%2Fbase..rkt%29._~23~25app%29%29).
